src/filesystem.hpp

```diff
@@ -39,7 +39,13 @@ namespace filesystem {
 using scoped_istream = std::unique_ptr<std::istream>;
 using scoped_ostream = std::unique_ptr<std::ostream>;
 
-typedef std::unique_ptr<SDL_RWops, void(*)(SDL_RWops*)> rwops_ptr;
+struct sdl_rwops_deleter
+{
+	void operator()(SDL_RWops*) const noexcept;
+};
+
+using rwops_ptr = std::unique_ptr<SDL_RWops, sdl_rwops_deleter>;
+static_assert(sizeof(rwops_ptr) == 8);
```

Comment:
Either this assert should have a useful message or it should be removed.

Reply:
The point being that it's not necessarily self-documenting - what's plainly obvious to you isn't necessarily so obvious to others.

Reply:
I see, you're right, I'm on it, will remove the assert ASAP (I feel it's better to remove it)

Reply:
the code has now changed and it's much better than what I did, should I still add a comment? should I remove the static_assert?

[here's a link to the latest version of it](https://github.com/wesnoth/wesnoth/commit/0e312fb74de309b2c2516e38b95b36ff2376222c)

Reply:
See https://github.com/wesnoth/wesnoth/commit/7bff35f991ee6bb4db4918a9165ad921a69086a7#r106243393
There is still no hint on why it's there so if there is no reason other than testing EBO then it should be removed.

Reply:
on it